concert/ext/ufo.py

```diff
@@ -152,7 +152,8 @@ def insert(self, array, node=None, index=0):
         else:
             self.ufo_buffers[node][index] = self.input_tasks[node][index].get_input_buffer()
 
-        self.ufo_buffers[node][index].copy_host_array(array.__array_interface__['data'][0])
+        if array is not None:
+            self.ufo_buffers[node][index].copy_host_array(array.__array_interface__['data'][0])
```

Comment:
I tried
```python
if array.dtype != np.float32:
    self.ufo_buffers[node][index].copy_host_array(2)
```

in order to delegate the float conversion to ufo but when I use more GPUs I get a segfault. Moreover, I cannot use the enumeration, because in python the prefix is stripped off and `Ufo.BufferDepth.16U` is invalid. I can work around this with `getattr(Ufo.BufferDepth, '16U')` but that's ripe for a facepalm... :smile:

Reply:
Actually, like we have `ufo_buffer_convert` now, we could in principle delegate this to the filters and write a GPU task which does that. The host buffer is always allocated for float in the beginning, right? In that case when I ask for a host buffer in my task an treat the array appropriately in OpenCL it should be possible. Or am I missing something except it's magicky?

Reply:
There is such a [task](https://github.com/ufo-kit/ufo-filters/blob/master/src/ufo-memory-in-task.c) but as of now it's doing conversion on the CPU (no idea, if it would actually be faster to do it on the GPU, I doubt it). And yes, the host buffer allocates float arrays *if* required.